File: nft-market-next/lib/api/auctions.ts
import { get } from './client';
import type { AuctionsFilterParams, AuctionsResponse } from '@/types';

/**
 * 获取拍卖列表
 * @param filters 过滤参数
 * @returns 拍卖列表
 */
export const getAuctions = async (filters?: AuctionsFilterParams): Promise<AuctionsResponse> => {
  const params = new URLSearchParams();
  
  if (filters) {
    if (filters.chainIds) {
      params.append('chain_ids', filters.chainIds.join(','));
    }
    if (filters.status) {
      params.append('status', filters.status);
    }
    if (filters.collectionAddrs) {
      params.append('collection_addrs', filters.collectionAddrs.join(','));
    }
    if (filters.userAddress) {
      params.append('user_address', filters.userAddress);
    }
    if (filters.priceMin) {
      params.append('price_min', filters.priceMin);
    }
    if (filters.priceMax) {
      params.append('price_max', filters.priceMax);
    }
    if (filters.page) {
      params.append('page', filters.page.toString());
    }
    if (filters.pageSize) {
      params.append('page_size', filters.pageSize.toString());
    }
  }
  
  const queryString = params.toString();
  return get(`/auctions${queryString ? `?${queryString}` : ''}`);
};

/**
 * 获取进行中的拍卖
 * @param limit 数量限制
 * @returns 进行中的拍卖列表
 */
export const getActiveAuctions = async (limit: number = 20) => {
  return get<AuctionsResponse>(`/auctions/active?limit=${limit}`);
};

/**
 * 获取即将结束的拍卖
 * @param limit 数量限制
 * @returns 即将结束的拍卖列表
 */
export const getEndingSoonAuctions = async (limit: number = 20) => {
  return get<AuctionsResponse>(`/auctions/ending-soon?limit=${limit}`);
};

File: EasySwapBackend/src/types/v1/auctions.go
package types

import "github.com/shopspring/decimal"

// AuctionInfo 拍卖信息
type AuctionInfo struct {
	AuctionID      string          `json:"auction_id"`      // 拍卖 ID
	ItemName       string          `json:"item_name"`       // 物品名称
	ItemImage      string          `json:"image_uri"`       // 图片 URL
	CollectionName string          `json:"collection_name"` // 集合名称
	CollectionAddr string          `json:"collection_addr"` // 集合地址
	TokenID        string          `json:"token_id"`        // Token ID
	Seller         string          `json:"seller"`          // 卖家
	HighestBidder  string          `json:"highest_bidder"`  // 最高出价者
	CurrentBid     decimal.Decimal `json:"current_bid"`     // 当前出价
	CurrentBidUSD  string          `json:"current_bid_usd"` // 当前出价 (USD)
	ReservePrice   decimal.Decimal `json:"reserve_price"`   // 保留价
	StartPrice     decimal.Decimal `json:"start_price"`     // 起始价
	BidCount       int64           `json:"bid_count"`       // 出价次数
	StartTime      int64           `json:"start_time"`      // 开始时间
	EndTime        int64           `json:"end_time"`        // 结束时间
	Status         string          `json:"status"`          // 状态：active, ended, cancelled
	ChainID        int             `json:"chain_id"`        // 链 ID
}

// AuctionsFilterParams 拍卖过滤参数
type AuctionsFilterParams struct {
	ChainID         []int    `json:"chain_ids"`        // 链 ID 列表
	Status          string   `json:"status"`           // 状态：active, ending_soon, new
	CollectionAddrs []string `json:"collection_addrs"` // 集合地址列表
	UserAddress     string   `json:"user_address"`     // 用户地址（创建或参与）
	PriceMin        string   `json:"price_min"`        // 最低价格
	PriceMax        string   `json:"price_max"`        // 最高价格
	Page            int      `json:"page"`             // 页码
	PageSize        int      `json:"page_size"`        // 每页数量
}

// AuctionsResponse 拍卖响应
type AuctionsResponse struct {
	Result []*AuctionInfo `json:"result"`
	Total  int64          `json:"total"`
}
